core/core.go

```diff
@@ -535,11 +552,23 @@ func Decrypt(jsonIn []byte) ([]byte, error) {
 		return jsonStatusError(err)
 	}
 
-	data, names, secure, err := crypt.Decrypt(s.Data, s.Name)
+	data, names, usages, secure, err := crypt.Decrypt(s.Data, s.Name)
 	if err != nil {
 		return jsonStatusError(err)
 	}
 
+	// a file must be marked as usable for "decrypt" before we will decrypt
+	// it for the user. If no usages are provided, we permit decryption
+	foundDecrypt := false
+	for _, usage := range usages {
+		if usage == "decrypt" {
+			foundDecrypt = true
```

Comment:
Ah good point; no reason not to break once we find "decrypt". Fixed in the next commit; and for the "ssh-sign-with" check as well. 

But the interface for SearchStrings is not convenient here; we'd need to first check whether the returned index is equal to the length of the list, then check if the string at that index is equal to "decrypt". Usages will be a very small list bounded in size by the number of API endpoints so it's probably not worth optimizing this.